lib/pos_mempool.go

```diff
@@ -580,3 +580,51 @@ func (mp *PosMempool) UpdateGlobalParams(globalParams *GlobalParamsEntry) {
 		glog.Errorf("PosMempool.UpdateGlobalParams: Problem refreshing mempool: %v", err)
 	}
 }
+
+// Implementation of the BackendMempool interface
+// These functions are used by the backend to interact with the mempool.
+
+func (mp *PosMempool) GetAugmentedUniversalView() (*UtxoView, error) {
+	if !mp.IsRunning() {
+		return nil, errors.Wrapf(MempoolErrorNotRunning, "PosMempool.GetAugmentedUniversalView: ")
+	}
+	newView, err := mp.readOnlyLatestBlockView.CopyUtxoView()
+	if err != nil {
+		return nil, errors.Wrapf(err, "PosMempool.GetAugmentedUniversalView: Problem copying utxo view")
+	}
+	return newView, nil
+}
+func (mp *PosMempool) GetAugmentedUtxoViewForPublicKey(pk []byte, optionalTx *MsgDeSoTxn) (*UtxoView, error) {
+	return mp.GetAugmentedUniversalView()
+}
+func (mp *PosMempool) CheckSpend(op UtxoKey) *MsgDeSoTxn {
+	panic("implement me")
+}
+
+func (mp *PosMempool) GetOrderedTransactions() []*MempoolTx {
+	mp.RLock()
+	defer mp.RUnlock()
+
+	if !mp.IsRunning() {
+		return nil
+	}
+	return mp.getTransactionsNoLock()
+}
+
+func (mp *PosMempool) IsTransactionInPool(txHash *BlockHash) bool {
+	// TODO: Do I need to hold a wider lock or nah?
+	mp.txnRegister.RLock()
+	defer mp.txnRegister.RUnlock()
```

Comment:
It's fine to hold the broader mempool lock since it's just a read lock. This way, we don't have to worry about edge cases where other data structures get updated but not the transaction register

```suggestion
	mp.RLock()
	defer mp.RUnlock()
```